test/github-package.test.js

```diff
@@ -8,617 +8,836 @@ import {fileExists, getTempDir} from '../lib/helpers';
+      beforeEach(async function() {
+        await contextUpdateAfter(githubPackage, () => githubPackage.activate());
+      });
 
-      const context = contextPool.getContext(workdirPath2);
-      assert.isTrue(context.isPresent());
-      assert.strictEqual(context.getRepository(), githubPackage.getActiveRepository());
-      assert.strictEqual(context.getResolutionProgress(), githubPackage.getActiveResolutionProgress());
-      assert.equal(githubPackage.getActiveWorkdir(), workdirPath2);
+      it('uses an undetermined repository context', function() {
+        assert.isTrue(githubPackage.getActiveRepository().isUndetermined());
+      });
     });
 
-    it('uses an active model from serialized state', async function() {
-      const [workdirPath1, workdirPath2, workdirPath3] = await Promise.all([
-        cloneRepository('three-files'),
-        cloneRepository('three-files'),
-        cloneRepository('three-files'),
-      ]);
-      project.setPaths([workdirPath1, workdirPath2, workdirPath3]);
+    context('with only 1 project', function() {
+      let workdirPath;
+      beforeEach(async function() {
+        workdirPath = await cloneRepository('three-files');
+        project.setPaths([workdirPath]);
 
-      await contextUpdateAfter(() => githubPackage.activate({
-        activeRepositoryPath: workdirPath2,
-      }));
+        await contextUpdateAfter(githubPackage, () => githubPackage.activate());
+      });
 
-      const context = contextPool.getContext(workdirPath2);
-      assert.isTrue(context.isPresent());
-      assert.strictEqual(context.getRepository(), githubPackage.getActiveRepository());
-      assert.strictEqual(context.getResolutionProgress(), githubPackage.getActiveResolutionProgress());
-      assert.equal(githubPackage.getActiveWorkdir(), workdirPath2);
+      it('uses the project\'s context', function() {
+        const context = contextPool.getContext(workdirPath);
+        assert.isTrue(context.isPresent());
+        assert.equal(githubPackage.getActiveWorkdir(), workdirPath);
+        assert.strictEqual(context.getRepository(), githubPackage.getActiveRepository());
+        assert.strictEqual(context.getResolutionProgress(), githubPackage.getActiveResolutionProgress());
+      });
     });
 
-    it('prefers the active model from an active pane item to serialized state', async function() {
-      const [workdirPath1, workdirPath2] = await Promise.all([
-        cloneRepository('three-files'),
-        cloneRepository('three-files'),
-      ]);
-      project.setPaths([workdirPath1, workdirPath2]);
-      await workspace.open(path.join(workdirPath2, 'b.txt'));
+    context('with only projects', function() {
+      let workdirPath1, workdirPath2, nonRepositoryPath;
+      beforeEach(async function() {
+        ([workdirPath1, workdirPath2, nonRepositoryPath] = await Promise.all([
+          cloneRepository('three-files'),
+          cloneRepository('three-files'),
+          getTempDir(),
+        ]));
```

Comment:
Oh neat, I don't think I've seen this trick to destructure into existing locals like this. Cool!